src/datalink/mod.rs

```diff
@@ -229,6 +229,15 @@ impl NetworkInterface {
     }
 }
 
+/// Represents an IP address and subnet mask
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub struct IpNetmask {
+    /// IP address
+    pub ip: IpAddr,
+    /// Subnet mask
+    pub netmask: Option<IpAddr>,
```

Comment:
Hrm. I know that when viewing output from `ifconfig` there are interfaces with no netmask, though that might just be because they have the default `0.0.0.0` et. al.? Happy to remove `Option` and set a default if it doesn't exist. I'm not sure what the underlying *Nix API assumes but the Windows API seems to think that there's always a subnet mask.